app-android/src/main/java/com/ustadmobile/port/android/view/PersonDetailFragment2.kt

```diff
@@ -79,31 +98,368 @@ class PersonDetailFragment2 : Fragment(){
     }
 }
 
+@Composable
+private fun PersonDetailScreen(
+    uiState: PersonDetailUiState = PersonDetailUiState(),
+    onClickDial: () -> Unit = {},
+    onClickSms: () -> Unit = {},
+    onClickEmail: () -> Unit = {},
+    onClickCreateAccount: () -> Unit = {},
+    onClickChangePassword: () -> Unit = {},
+    onClickManageParentalConsent: () -> Unit = {},
+    onClickChat: () -> Unit = {},
+    onClickClazz: () -> Unit = {}
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .verticalScroll(rememberScrollState())
+    )  {
+
+        Image(
+            painter = painterResource(id = R.drawable.ic_person_black_24dp),
+            contentDescription = null,
+            modifier = Modifier
+                .height(256.dp)
+                .fillMaxWidth())
+
+        QuickActionBar(
+            uiState,
+            onClickDial,
+            onClickSms,
+            onClickEmail,
+            onClickCreateAccount,
+            onClickChangePassword,
+            onClickManageParentalConsent,
+            onClickChat)
+
+        Divider(color = Color.LightGray, thickness = 1.dp)
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        Text(stringResource(R.string.basic_details),
+            style = Typography.h4,
+            modifier = Modifier.padding(8.dp))
+
+        DetailFeilds(uiState)
+
+        Divider(color = Color.LightGray, thickness = 1.dp)
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        Text(stringResource(R.string.contact_details),
+            style = Typography.h4,
+            modifier = Modifier.padding(8.dp))
+
+        ContactDetails(uiState,
+            onClickDial,
+            onClickSms,
+            onClickEmail)
 
+        Divider(color = Color.LightGray, thickness = 1.dp)
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        Text(stringResource(R.string.classes),
+            style = Typography.h4,
+            modifier = Modifier.padding(8.dp))
+
+        Classes(uiState.clazzes, onClickClazz)
+    }
+}
 
 @Composable
-fun PersonDetailScreen(
-    person: Person? = null,
-    onClickChangeAccount: () -> Unit = {},
+private fun QuickActionBar(
+    uiState: PersonDetailUiState,
+    onClickDial: () -> Unit = {},
+    onClickSms: () -> Unit = {},
+    onClickEmail: () -> Unit = {},
+    onClickCreateAccount: () -> Unit = {},
+    onClickChangePassword: () -> Unit = {},
+    onClickManageParentalConsent: () -> Unit = {},
+    onClickChat: () -> Unit = {},
 ) {
-    Text(person?.firstNames ?: "No Name")
+    Row(
+        modifier = Modifier.horizontalScroll(rememberScrollState())
+    ) {
+
+        if (!uiState.person?.phoneNum.isNullOrEmpty()){
```

Comment:
These properties should be put on the UiState. Then they are shared between Javascript and Android. I have pushed an update for the UiState.